registry/registry_ui.go

```diff
@@ -22,16 +22,131 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"reflect"
 	"strconv"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/gin-gonic/gin"
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/param"
+	"github.com/pelicanplatform/pelican/web_ui"
+	log "github.com/sirupsen/logrus"
 )
 
-type listNamespaceRequest struct {
-	ServerType string `form:"server_type"`
+type (
+	listNamespaceRequest struct {
+		ServerType string `form:"server_type"`
+	}
+
+	registrationFieldType string
+	registrationField     struct {
+		Name     string                `json:"name"`
+		Type     registrationFieldType `json:"type"`
+		Required bool                  `json:"required"`
+		Options  []interface{}         `json:"options"`
+	}
+)
+
+const (
+	String   registrationFieldType = "string"
+	Int      registrationFieldType = "int"
+	Enum     registrationFieldType = "enum"
+	DateTime registrationFieldType = "datetime"
+)
+
+var (
+	registrationFields        []registrationField
+	setRegistrationFieldsOnce sync.Once
+)
+
+func init() {
+	setRegistrationFieldsOnce.Do(func() {
```

Comment:
I don't think the `once` function is necessary here.  The `init()` function is guaranteed to be invoked once.

`once` objects are useful when you want to do the initialization lazily -- i.e., cases when it's expensive to do the computation so you only want it done when the result is actually needed, not every time the server starts.